bin/handler-slack-multichannel.rb

```diff
@@ -256,31 +263,44 @@ def post_data(notice, channel)
     http.use_ssl = true
 
     begin
-      req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}")
-      if payload_template.nil?
-        text = slack_surround ? slack_surround + notice + slack_surround : notice
-        req.body = payload(text, channel).to_json
+      # Implement a retry strategy, with 5 tries max and a timeout of 10 seconds each
+      tries ||= 5
+      Timeout.timeout(10) do
+        begin
+          req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}")
+          if payload_template.nil?
+            text = slack_surround ? slack_surround + notice + slack_surround : notice
```

Comment:
we can use string interpolation over string concatenation here which is faster and requires less memory as string interpolation creates 1 string where concatenation creates `N+1` strings where `N` is the number of strings to concatenate.

```suggestion
            text = slack_surround ? "#{slack_surround}#{notice}#{slack_surround} : notice
```

Reply:
Agree, will change this and all the other to interpolation. This line only got indentation changed however, so I did not touch it. But since I agree with the suggestion, I will add it regardless.